kubernetes/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterScheduler.scala

```diff
@@ -225,22 +156,19 @@ private[spark] class KubernetesClusterScheduler(conf: SparkConf)
   }
 
   def setupKubernetesClient(): KubernetesClient = {
-      val sparkMaster = new java.net.URI(conf.get("spark.master"))
-      if (sparkMaster.getHost() == "default") {
-        return new DefaultKubernetesClient()
-      } else {
-        var config = new ConfigBuilder().withMasterUrl(sparkMaster.getHost()).build
-        var client = new DefaultKubernetesClient(config)
-        return client
-      }
-  }
+    val sparkHost = new java.net.URI(conf.get("spark.master")).getHost()
 
-  def getNamespace(): String = {
-    var kubernetesNamespace = System.getenv("K8S_NAMESPACE")
-    if (kubernetesNamespace == null) {
-      kubernetesNamespace = "default"
+    var config = new ConfigBuilder().withNamespace(nameSpace)
+    if (sparkHost != "default") {
```

Comment:
if you set it to "default" it will attach to whichever cluster your local `~/.kube/` environment indicates.